### ISSUE TITLE
[Feature request] waterbox melonds

### ISSUE THREAD

**Issue by InfamousKnight** [TASVideos/BizHawk#2943]
While I did find desync problems with melonds, consistently reproducing them and fixing them sounds like a pain. So waterboxing it would provide piece of mind while tasing.

I would use it for driving games.

**Comment by CasualPokePlayer** [comment]
There are some things to consider if doing this:

1. Threads are (effectively) gone. Threads can be used by the software renderer (which in practice should be non-problematic determinism wise since that's just rendering), and would provide a good performance boost with multi-core CPUs. This boost is gone from waterboxing due to the lack of actual threads.

2. OpenGL... yeah I have severe doubts this can even be implemented if going the wb approach (I'd be very surprised if it's possible to access opengl shit within wb). Still, this is the less accurate option and is more for possible performance boosts (or on the flip side goodsizers).

3. No idea if JIT would work. I'd imagine that'd be an instant nope on wb but I wouldn't really know.

Upstream states seem mostly fine, although it is concerning at some sections, and wb is an effective be all fix any wouldn't need to deal with any upstream state issues. Still, pretty much means a "well fuck casual users", which granted probably should just use upstream in the first place.

**Comment by InfamousKnight** [comment]
Yeah lol. Might as well use melonds upstream if being used casually. Would much rather tas with no worry of some desync happening along the way and then fight with the emulator to continue.
